src/article/shared/AddFigurePanelCommand.js

```diff
@@ -0,0 +1,39 @@
+import { Command } from 'substance'
+import { findParentByType } from './nodeHelpers'
+
+export default class AddFigurePanelCommand extends Command {
+  getCommandState (params, context) {
+    return {
+      disabled: this.isDisabled(params)
+    }
+  }
+
+  execute (params, context) {
+    const collection = this._getCollection(params, context)
+    const files = params.files
+    collection.addItem({})
```

Comment:
I think you should not add twice. I.e. add only one new item to the current panel collection.